### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by Roker2**
- Fix warnings
- Update RU translation for `libcore`
- Add using const references
- Use properties as variables
- Some optimizations and improvements

### CODE REVIEW

**Comment by Roker2** [comment]
I didn't check how to works the app with this changes, I can check it later

File: libcore/src/KeyboardLayout.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ class CORE_LIB_EXPORT KeyboardLayout : public QObject
 		};
 		Q_ENUM(Finger)
 
-		enum Row
+		enum class Row
```
Comment (adazem009) [inline libcore/src/KeyboardLayout.h:78]: I'd like to avoid using `enum class` because QML doesn't support it.

In QML, the enum can be accessed like this:
```qml
KeyboardLayout.Row_A
```
assuming that `KeyboardLayout` is an **instance** of the `KeyboardLayout` C++ class (because enums won't work without an instance).

I can't access the enum like this from QML:
```qml
KeyboardLayout.Row.Row_A
```

Using `enum class` changes it only for C++.

File: libcore/src/HistoryParser.cpp [hunk 2]
```diff
@@ -21,6 +21,11 @@
 #include "HistoryParser.h"
 #include "FileUtils.h"
 
+const QString HistoryParser::historyFile = "/history.json";
+const QString HistoryParser::speedProperty = "speed";
```
Comment (adazem009) [inline libcore/src/HistoryParser.cpp:26]: This could be also done with the `AddonItemModel` class (in `app/src/AddonItemModel.cpp`) which reads the JSON file of a downloaded addon, and the `AddonManager` class (in `app/src/AddonManager.cpp`) which reads the JSON files of installed addons.

File: libcore/translations/libcore_ru_RU.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@
     <message>
         <location filename="../src/AppMenuBar.cpp" line="102"/>
         <source>Quit</source>
-        <translation type="unfinished"></translation>
+        <translation>Выйти</translation>
```
Comment (adazem009) [inline libcore/translations/libcore_ru_RU.ts:24]: I'd move this to a separate commit.

File: libcore/src/HistoryParser.cpp (hunk 2 above)
Comment (Roker2) [reply libcore/src/HistoryParser.cpp:26]: `AddonItemModel` - done

Comment (Roker2) [reply libcore/src/HistoryParser.cpp:26]: `AddonManager` - done

File: libcore/src/KeyboardLayout.cpp [hunk 4]
```diff
@@ -3,6 +3,7 @@
  * This file is part of Open-Typer
  *
  * Copyright (C) 2023 - adazem009
+ * Copyright (C) 2023 - Roker2
```
Comment (Roker2) [inline libcore/src/KeyboardLayout.cpp:6]: Need to remove it, I forgot it

**Review by adazem009 — APPROVED** [review]
There are still some unfinished translations remaining, but I'll keep them for future pull requests.

File: app/src/AddonManager.cpp [hunk 5]
```diff
@@ -29,6 +30,13 @@
 
 AddonManager globalAddonManager;
 
+const QString AddonManager::addonModelNameProperty = "name";
+const QString AddonManager::addonModelDescriptionProperty = "description";
+const QString AddonManager::addonModelIconFileNameProperty = "iconFileName";
+const QString AddonManager::addonModelRepositoryUrlProperty = "repositoryUrl";
+const QString AddonManager::addonModelVersionProperty = "version";
+const QString AddonManager::addonModelQtMajorProperty = "qtMajor";
```
Comment (adazem009) [inline app/src/AddonManager.cpp:38]:
```suggestion
const QString AddonManager::addonModelQtMajorProperty = "qtMajor";

AddonManager globalAddonManager;
```
One detail before I can merge this.
globalAddonManager should be defined after the property strings because the application will crash when they're accessed from the AddonManager constructor.

File: app/src/AddonManager.cpp [hunk 6]
```diff
@@ -29,6 +30,13 @@
 
 AddonManager globalAddonManager;
```
Comment (adazem009) [inline app/src/AddonManager.cpp:31]:
```suggestion

```

File: app/src/AddonManager.cpp (hunk 5 above)
Comment (Roker2) [reply app/src/AddonManager.cpp:38]: It didn't help, crash exists.

**Review by adazem009 — APPROVED** [review]
